storage/client_status_test.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram.gruneir@gmail.com)
+
+package storage_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util/hlc"
+	"github.com/cockroachdb/cockroach/util/leaktest"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// compareStoreStatus ensures that the actual store status for the passed in
+// store is updated correctly. It checks that the StoreID, NodeID and
+// RangeCount are exactly correct and that the bytes and counts for Live, Key
+// and Val are at least the expected value.  The latest actual stats are
+// returned.
+func compareStoreStatus(t *testing.T, store *storage.Store, expectedStoreStatus *proto.StoreStatus, testNumber int) *proto.StoreStatus {
+	storeStatusKey := engine.StoreStatusKey(int32(store.Ident.StoreID))
+	gArgs, gReply := getArgs(storeStatusKey, 1, store.Ident.StoreID)
+	if err := store.ExecuteCmd(gArgs, gReply); err != nil {
+		t.Fatalf("%v: failure getting store status: %s", testNumber, err)
+	}
+	if gReply.Value == nil {
+		t.Errorf("%v: could not find store status at: %s", testNumber, storeStatusKey)
+	}
+	storeStatus := &proto.StoreStatus{}
+	if err := gogoproto.Unmarshal(gReply.Value.GetBytes(), storeStatus); err != nil {
+		t.Fatalf("%v: could not unmarshal store status: %+v", testNumber, gReply)
+	}
+	if expectedStoreStatus.StoreID != storeStatus.StoreID {
```

Comment:
you probably had a reason, but on first sight this looks like a very manual way to get what `reflect.DeepEqual` does.